### ISSUE TITLE
hostName field reported by Tracee should be 64 characters long

### ISSUE THREAD

**Issue by hangrymuppet** [aquasecurity/tracee#3840]
Version: 0.19.0

An example of this is already in the Tracee repo [here](https://github.com/aquasecurity/tracee/blob/ccb2c11ee70e28a845afbd4410240c11aa3c98fd/deploy/helm/tracee/templates/NOTES.txt#L7)

Most of the tests in the project have hostnames that are under 15 characters which is why I think this was not caught before.

**Comment by yanivagman** [comment]
Yes, this is "by design" - tracee uses a fixed-size context for each event originated from eBPF code (~132 bytes).
To save space, we limit the size of the UTS name to 15 chars, as can be seen here: https://github.com/aquasecurity/tracee/blob/main/pkg/ebpf/c/common/context.h#L71

Since we now store information about threads/processes in userspace, it might be possible to include the full hostname by saving it there. This task requires some work, including considering the possibility of the UTS name being modified by the process after its initiation.

**Comment by yanivagman** [comment]
task name (aka "comm") in Linux is limited to 16 characters - https://elixir.bootlin.com/linux/v6.7.5/source/include/linux/sched.h#L291
So we don't trim it, this is just the way it is. 

UTS name, on the other hand, is limited in the kernel to 64 chars, so it is a different case.
I updated the issue to make it a feature request.

Regarding "process name", I think the full name you are looking for is actually the binary name. This field will have the full name as part of the path, and we plan to add it to Tracee soon (see context->process->executable here: https://github.com/aquasecurity/tracee/issues/2870)

BTW, we already keep all the information about processes in userspace today. This is done in here: https://github.com/aquasecurity/tracee/tree/main/pkg/proctree

**Comment by yanivagman** [comment]
hostName is no longer part of tracee event output:
https://aquasecurity.github.io/tracee/dev/docs/outputs/event-structure/
